### DISCUSSION TITLE
Is lage a replacement for or complimentary to heft?

### DISCUSSION THREAD

**Discussion by smolinari** [microsoft/lage#176] [General]
I'm relatively new to the Rush ecosystem and ran into this little gem of a package and was wondering is it a replacement for Heft or is it something complimentary to Heft? I might have my understanding of concepts crossed too, so I figured I'd ask. 

Scott

**Comment by kenotron** [comment]
Sorry, @smolinari - as the owner of this package, I have NOT yet established a good official place to do discussions. But, I did try to start it here, so here goes!

Rush is a Swiss Army Knife, if you like to think of it this way: it combines all aspects of monorepo management / build / publish / etc. in one tool. lage is different: it was mainly there to onboard repos that are already with code - it only requires a single devDep in the root + a little config.

Heft is something you would place inside each package.json file. As such, Rush *or* lage would launch those package.json scripts for you. So long answer: yes, it is totally complementary. I find that it has some niceties unique in that it can reuse TS projects, etc.
